opensrp-reveal/src/main/java/org/smartregister/reveal/sync/RevealClientProcessor.java

```diff
@@ -92,14 +92,16 @@ public void processClient(List<EventClient> eventClients, boolean localEvents) {
             }
         }
 
+
         if (hasSynchedEventsInTarget) {
             Intent intent = new Intent(STRUCTURE_TASK_SYNCHED);
             LocalBroadcastManager.getInstance(getContext()).sendBroadcast(intent);
         }
 
     }
 
-    private void processSprayEvent(Event event, ClientClassification clientClassification, boolean localEvents) {
+    private String processSprayEvent(Event event, ClientClassification clientClassification, boolean localEvents) {
```

Comment:
Logic to extract operationalAreaId would better be in it's own function?

Reply:
The method has a for loop and they have been combined because of optimization. Avoiding 2 for loops